crypto/x509/v3_purp.c

```diff
@@ -361,21 +358,25 @@ static void x509v3_cache_extensions(X509 *x)
     EXTENDED_KEY_USAGE *extusage;
     X509_EXTENSION *ex;
     int i;
+    EVP_MD *sha1;
 
 #ifdef tsan_ld_acq
     /* fast lock-free check, see end of the function for details. */
     if (tsan_ld_acq((TSAN_QUALIFIER int *)&x->ex_cached))
-        return;
+        return (x->ex_flags & EXFLAG_INVALID) == 0;
```

Comment:
I wonder if calling this bit out separately as an atomic-int field is worthwhile.

Reply:
The `ex_flags` can be obtained by application code using `X509_get_extension_flags()`, so for backwards compat reasons we'd still have to keep the `EXFLAG_INVALID` flag anyway. If we were designing this from scratch then what you say might make sense - but probably not now.

Reply:
You can keep the API, just merge both fields as the return from the "get-flags" method.

I am slightly concerned about the thread-safety of non-atomic (bitwise) operations, but oh well.

Reply:
> You can keep the API, just merge both fields as the return from the "get-flags" method.

We could do, but I don't think its worth it.